ct-ng build for arm-unknown-linux-gnueabi fails when installing ncurses:

I've got some trouble building `crosstool-ng 1.28.0` for Linux / x86_64 (host) with `arm-unknown-linux-gnueabihf` as a target: After letting `ct-ng build` run for some half-minute, when trying to install `ncurses`, it spits out this error message:

```ini
[ERROR]    checking for C compiler default output... configure: error: C compiler cannot create executables
```

Here is the full output:
```ini
╭─jacek@epica /opt/crosstool-ng-1.28.0  
╰─➤  ./ct-ng build     
[INFO ]  Performing some trivial sanity checks
[WARN ]  Number of open files 1024 may not be sufficient to build the toolchain; increasing to 2048
[INFO ]  Build started 20251206.180257
[INFO ]  Building environment variables
[EXTRA]  Preparing working directories
[EXTRA]  Installing user-supplied crosstool-NG configuration
[EXTRA]  =================================================================
[EXTRA]  Dumping internal crosstool-NG configuration
[EXTRA]    Building a toolchain for:
[EXTRA]      build  = x86_64-pc-linux-gnu
[EXTRA]      host   = x86_64-pc-linux-gnu
[EXTRA]      target = arm-unknown-linux-gnueabihf
[EXTRA]  Dumping internal crosstool-NG configuration: done in 0.07s (at 00:02)
[INFO ]  =================================================================
[INFO ]  Retrieving needed toolchain components' tarballs
[INFO ]  Retrieving needed toolchain components' tarballs: done in 0.68s (at 00:03)
[INFO ]  =================================================================
[INFO ]  Extracting and patching toolchain components
[INFO ]  Extracting and patching toolchain components: done in 0.75s (at 00:04)
[INFO ]  =================================================================
[INFO ]  Installing make for build
[EXTRA]    Configuring make
[EXTRA]    Building make
[EXTRA]    Installing make
[INFO ]  Installing make for build: done in 11.23s (at 00:15)
[INFO ]  =================================================================
[INFO ]  Installing automake for build
[EXTRA]    Configuring automake
[EXTRA]    Building automake
[EXTRA]    Installing automake
[INFO ]  Installing automake for build: done in 2.89s (at 00:18)
[INFO ]  =================================================================
[INFO ]  Installing ncurses for build
[EXTRA]    Configuring ncurses
[ERROR]    checking for C compiler default output... configure: error: C compiler cannot create executables
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing ncurses for build'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@378]
[ERROR]  >>        called from: do_ncurses_backend[scripts/build/companion_libs/220-ncurses.sh@158]
[ERROR]  >>        called from: do_ncurses_for_build[scripts/build/companion_libs/220-ncurses.sh@32]
[ERROR]  >>        called from: do_companion_libs_for_build[scripts/build/companion_libs.sh@29]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@712]
[ERROR]  >>
[ERROR]  (elapsed: 0:17.60)
[00:18] / gmake: *** [ct-ng:261: build] Fehler 77
```

As I can't find anything on the Web about this kind of error, may I hereby open an issue. Is `ct-ng build` maybe using the wrong C compiler?

I don't need ncurses at all in my project (headless app), but in `ct-ng menuconfig` there is apparently no possibility to disable it.

Are these the files you are looking for:

[build.log](https://github.com/user-attachments/files/24003496/build.log)

[config.log](https://github.com/user-attachments/files/24003497/config.log)

In the `config.log` I can find two messages in lines 313 thru 320, which make me very nervous:

```
gcc: error: unrecognized command line option '-V'
gcc: fatal error: no input files
compilation terminated.
configure:4381: $? = 1
configure:4370: gcc -qversion >&5
gcc: error: unrecognized command line option '-qversion'; did you mean '--version'?
gcc: fatal error: no input files
compilation terminated.
```

And slightly later in lines 335 thru 342:

```
configure:5533: gcc -E  conftest.c
conftest.c:11:10: fatal error: ac_nonexistent.h: No such file or directory
 #include <ac_nonexistent.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
configure:5533: $? = 1
configure: failed program was:
| /* confdefs.h */
```

--- a/scripts/build/companion_libs/100-gmp.sh
+++ b/scripts/build/companion_libs/100-gmp.sh
@@ -77,7 +77,7 @@ do_gmp_for_target() {
             prefix="/usr"
             ;;
     esac
-    gmp_opts+=( "cflags=${CT_ALL_TARGET_CFLAGS}" )
+    gmp_opts+=( "cflags=${CT_ALL_TARGET_CFLAGS} ${CT_GMP_EXTRA_CFLAGS}" )
     gmp_opts+=( "prefix=${prefix}" )
     gmp_opts+=( "destdir=${CT_SYSROOT_DIR}" )
     gmp_opts+=( "shared=${CT_SHARED_LIBS}" )
@@ -121,8 +121,6 @@ do_gmp_backend() {
     if [ "${CT_CC_LANG_JIT}" = "y" ]; then
         extra_config+=("--with-pic")
     fi
-
-    cflags+=" ${CT_GMP_EXTRA_CFLAGS}"
 
     # GMP's configure script doesn't respect the host parameter
     # when not cross-compiling, ie when build == host so set

--- a/scripts/build/companion_libs/220-ncurses.sh
+++ b/scripts/build/companion_libs/220-ncurses.sh
@@ -97,7 +97,7 @@ do_ncurses_for_target() {
                        prefix="${prefix}" \
                        destdir="${CT_SYSROOT_DIR}" \
                        shared="${CT_SHARED_LIBS}" \
-                       cflags="${CT_ALL_TARGET_CFLAGS}" \
+                       cflags="${CT_ALL_TARGET_CFLAGS} ${CT_NCURSES_EXTRA_CFLAGS}" \
                        "${opts[@]}"
     CT_Popd
     CT_EndStep
@@ -151,8 +151,6 @@ do_ncurses_backend() {
     if [ "${shared}" = "y" ]; then
         ncurses_opts+=("--with-shared")
     fi
-
-    cflags+=" ${CT_NCURSES_EXTRA_CFLAGS}"
 
     CT_DoLog EXTRA "Configuring ncurses"
     CT_DoExecLog CFG                                                    \

